docs/Technical-Documentation/tech-memos/priotitized-errors/prioritized-errors.md

```diff
@@ -0,0 +1,101 @@
+
+### Memory Efficient Report Generation
+As previously mentioned in the #background section, the `get_xls_serialized_file` introduces a method to serialize parser errors into a XLSX that requires the entire queryset of parser errors to be brought into memory. Because these querysets can be very large, having them in memory regularly kills Gunicorn workers with an OOM error. To remedy the issue, this tech memo suggests updating `get_xls_serialized_file` to not use Django model serializers and instead leverage the power of Django querysets and pagination. To accomplish this, instead of passing a JSON serialized querset to `get_xls_serialized_file`, a standard (un-evaluated) queryset should be passed. Then, the body of the `get_xls_serialized_file` function should be updated appropriately to use a queryset object instead of a JSON object to generate the XLSX spreadsheet. The updates should also include paginating the queryset to avoid bringing the entirety of the queryset into memory at any one time. The code snippet below provides an example of paginating the queryset and writing the appropriate fields of each entry to the XLSX report.
+
+```python
+paginator = Paginator(parser_errors, settings.BULK_CREATE_BATCH_SIZE)
+row_idx = 6
+for page in paginator:
+    for record in page.object_list:
+        rpt_month_year = str(getattr(record, 'rpt_month_year', None))
+        fields_json = getattr(record, 'fields_json', {})
+
+        worksheet.write(row_idx, 0, record.case_number)
+        worksheet.write(row_idx, 1, rpt_month_year[:4])
+        worksheet.write(row_idx, 2, calendar.month_name[int(rpt_month_year[4:])] if rpt_month_year[4:] else None)
+        worksheet.write(row_idx, 3, format_error_msg(record.error_message, fields_json))
+        worksheet.write(row_idx, 4, record.item_number)
+        worksheet.write(row_idx, 5, friendly_names(fields_json))
+        worksheet.write(row_idx, 6, internal_names(fields_json))
+        worksheet.write(row_idx, 7, record.row_number)
+        worksheet.write(row_idx, 8, str(ParserErrorCategoryChoices(record.error_type).label))
+```
+
+The three helper functions: `format_error_msg`, `friendly_names`, `internal_names` used to write the appropriate fields can be seen below.
+
+```python
+def format_error_msg(error_msg, fields_json):
+    """Format error message."""
+    for key, value in fields_json['friendly_name'].items():
+        error_msg = error_msg.replace(key, value) if value else error_msg
+    return error_msg
+
+
+def friendly_names(fields_json):
+    """Return comma separated string of friendly names."""
+    return ','.join([i for i in fields_json['friendly_name'].values()])
+
+
+def internal_names(fields_json):
+    """Return comma separated string of internal names."""
+    return ','.join([i for i in fields_json['friendly_name'].keys()])
+```
+
+### Prioritized/Critical Errors
```

Comment:
one thing we missed capturing in the onenote is that Aggregate and Stratum files with any 2: Record Value Invalid and 3: Record value consistency errors should be on critical worksheet.

Reply:
That is not taken into account in the code either. I will need to add that functionality or we can write another ticket to add it.

Reply:
#3341